impala/hiveserver2.py

```diff
@@ -1024,11 +1024,11 @@ def close(self):
         req = TCloseSessionReq(sessionHandle=self.handle)
         self._rpc('CloseSession', req)
 
-    def execute(self, statement, configuration=None, async=False):
+    def execute(self, statement, configuration=None, run_async=False):
```

Comment:
If you really want to be thorough here you can maintain backwards compatibility by going through a deprecation cycle.

You would do this by setting the default value of `run_async` to `None`, adding `**kwargs` to the method signature, checking if `run_async` is not `None` and using that if so else use `kwargs.get('async', False)`. If you do decide to do this then you should also check if `'async' in kwargs` and give a warning to users that their code needs to be updated to use `run_async` because in the next release you will not accept this keyword argument.

Reply:
@wesm Actually, are you aware of whether this method `HS2Session.execute` is part of the public API? It looks like it might not be in which case we don't have to handle backwards compatibility here.

Reply:
After digging around a bit I don't think this method is part of the public API. Users go through `connect -> HiveServer2Connection -> .cursor() -> HiveServer2Cursor -> execute/execute_async` and never see the `HS2Session.execute` calls.